modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/RadioFilter.js

```diff
@@ -103,6 +103,7 @@ function RadioFilter({id, items, selectedData, setFilter}) {
 							};
 
 							setFilter({
```

Comment:
This is just a nit

But traversing setState around components makes me feel worried. We can use a callback called onFilterChange and make the set state in a parent component. https://reactjs.org/docs/lifting-state-up.html

Reply:
Hmm, interesting point. This may just be a naming convention we are talking about here. If you squint, you could see `setFilter` as a callback. Top component, `FrontendDataSet`, holds the state. It just happens that `onFilterChange` only does `setFilter`, so we might as well pass it down as `setFilter`.

In the same way, if it's more than just a setter, we are passing down a custom function. For example, this [`selectAllItems`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L428-L430) or this [`createInlineItem`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L679).

Something to have in mind is that we are dealing with a context prop here, not a simple component prop. Passing down a function may make more sense in this case. If we are passing down callbacks for different events, we may end up with a lot of callbacks invoking the same function.